### ISSUE TITLE
Rclone encrypted mount reading files outside the specified path

### ISSUE THREAD

**Issue by ArabCoders** [rclone/rclone#6771]
#### What is the problem you are having with rclone?

I have an encrypted google drive backend, it's working perfectly for god knows how long, i upgraded recently to `v1.61.1`, and i begin to receive unexpected error, log is filled with something like

```
Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
```

It's only one file that shows the problem and this file isn't even part of the `/gdrive/encrypt`, i suspect because it's reading on changeNotify it's reading all files that are part of the google drive not just the ones namespaced under `/gdrive/encrypt`. because i did `lsjson gnocache: --recursive` on both the `encrypted` and `unencrypted` endpoint. and the reported filename isn't there, however when i go directly to google drive, and search for the file it's a file that was shared with me and it's frequently being updated as such, i think rclone is listening on all files not just what it's supposed to and ignore the others. while i think it's not problem it's might leak things which aren't supposed to. in my case it's non issues as i don't really use the drive much except for rclone.

#### Run the command 'rclone version' and share the full output of the command.

rclone v1.61.1
- os/version: ubuntu 22.04 (64 bit)
- os/kernel: 5.15.0-60-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.19.4
- go/linking: static
- go/tags: none

####  Which cloud storage system are you using? (eg Google Drive)

Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)  

```
/usr/bin/rclone mount gnocache: /mnt/google --allow-other --dir-cache-time 1000h --log-level NOTICE --syslog --poll-interval 15s --user-agent random_agent --rc --rc-addr=127.0.0.1:8070 --rc-no-auth --cache-dir=/rcache --cache-db-purge --vfs-cache-mode full --vfs-cache-max-size 50G --vfs-cache-max-age 6h --vfs-cache-poll-interval 5m --drive-pacer-min-sleep 10ms --drive-pacer-burst 1000 --attr-timeout 5000h --vfs-read-chunk-size 1M --bwlimit-file 32M
```

#### The rclone config contents with secrets removed.  
```
[gdrive]
type = drive
scope = drive
token = snip
client_id = snip
client_secret = snip
root_folder_id = snip

[gnocache]
type = crypt
remote = gdrive:/gdrive/crypt
filename_encryption = standard
directory_name_encryption = true
password = snip
password2 = snip
```

#### A log from the command with the `-vv` flag  

```
Feb 15 09:13:46 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 15 15:48:46 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 15 16:48:31 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 15 17:03:01 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 15 17:06:01 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 15 17:13:46 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 15 22:50:46 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 16 02:28:01 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 16 18:20:31 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
Feb 16 18:31:46 main rclone[1293]: Encrypted drive 'gnocache:': ChangeNotify was unable to decrypt "Scene Naming (Responses)": illegal base32 data at input byte 5
```

as per the discussion in https://forum.rclone.org/t/rclone-encrypted-mount-reading-files-outside-the-directory/ it was suggested that i open a bug report here.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
These are most likely change notifications for shared files which are unencrypted as you suspected.

Assuming that is the case, this should be easy to fix.

Can you try this please?

[v1.62.0-beta.6743.35242fafe.fix-6771-drive-changes](https://beta.rclone.org/branch/fix-6771-drive-changes/v1.62.0-beta.6743.35242fafe.fix-6771-drive-changes/) on branch [fix-6771-drive-changes](https://github.com/rclone/rclone/tree/fix-6771-drive-changes) (uploaded in 15-30 mins)

**Comment by ArabCoders** [comment]
Sorry for late response, i have family health matter to attend to. Once i have free time i will report back.
